tests/e2e/customsuites/directorypermissions.go

```diff
@@ -586,4 +587,323 @@ func (t *s3CSIDirectoryPermissionsTestSuite) DefineTests(driver storageframework
+		expectedMode := strings.TrimSpace(statMode)
+		gomega.Expect(expectedMode).To(gomega.Equal("700"))
+
+		// Confirm access() matches: we expect R/W/X to succeed
+		gomega.Expect(errR).NotTo(gomega.HaveOccurred(), "access(R_OK) failed unexpectedly")
+		gomega.Expect(errW).NotTo(gomega.HaveOccurred(), "access(W_OK) failed unexpectedly")
+		gomega.Expect(errX).NotTo(gomega.HaveOccurred(), "access(X_OK) failed unexpectedly")
+	})
+
+	// --------------------------------------------------------------------
+	// Directory Name Edge Cases (Full Path Test)
+	//
+	// This test ensures that Mountpoint-S3 can handle directories with
+	// complex names (Unicode, long names, special characters) and that
+	// full read/write functionality inside these directories works.
+	//
+	// Why? S3 key encoding applies to directory paths just like files,
+	// and we must ensure the driver isn't mishandling edge cases.
+	//
+	// Test steps for each case:
+	//
+	// ┌─────────────────────────────────────────────────────────────┐
+	// │ 1. mkdir <dir>                                              │
+	// │ 2. stat <dir>                                               │
+	// │ 3. touch <dir>/.marker  (store 'ok' text in the file)       │
+	// │ 4. cat <dir>/.marker    (read it back to confirm)           │
+	// │                                                             │
+	// │ Directory cases:                                            │
+	// │  - unicode-🚀🌟                                             │
+	// │  - long-(200 chars)                                         │
+	// │  - special-!@#$%^&()[]{}=+,~                                │
+	// └─────────────────────────────────────────────────────────────┘
+	//
+	// Expected:
+	// - mkdir, stat, write, and read all succeed.
+	// - Marker file content is correct ("ok").
+	// - No path encoding errors.
+	//
+	// This tests full end-to-end path handling, not just directory creation.
+	//
+	ginkgo.It("should support unicode / long / special‑char directory names with file storage", func(ctx context.Context) {
+		res := createVolume(ctx, testRegistry.config, pattern,
+			DefaultNonRootUser, DefaultNonRootGroup, "0700")
+
+		pod, err := CreatePodWithVolumeAndSecurity(ctx, f, res.Pvc, "",
+			DefaultNonRootUser, DefaultNonRootGroup)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer e2epod.DeletePodWithWait(ctx, f.ClientSet, pod)
+
+		dirs := []string{
+			"/mnt/volume1/unicode-🚀🌟",
+			"/mnt/volume1/long-" + strings.Repeat("x", 200),
+			"/mnt/volume1/special-!@#$%^&()[]{}=+,~",
+		}
+
+		for _, dir := range dirs {
+			ginkgo.By(fmt.Sprintf("Creating and testing directory: %s", dir))
+
+			// 1: Create directory - replacing with helper
+			CreateDirInPod(f, pod, dir)
+
+			// 2: Stat the directory
+			_, stderr, err := e2evolume.PodExec(f, pod, fmt.Sprintf("stat '%s'", dir))
+			framework.ExpectNoError(err, fmt.Sprintf("stat failed for %s: %s", dir, stderr))
+
+			// 3: Create marker file inside - replacing with helper
```

Comment:
` - replacing with helper`

Reply:
I left these for changes, will rmeove them

Reply:
addressed resolving